src/appleseed.studio/utility/miscellaneous.cpp

```diff
@@ -324,6 +324,21 @@ QString get_save_filename(
             QDir::toNativeSeparators(QFileInfo(filepath).path()));
     }
 
+    QFileInfo finfo(filepath);
+
+    // Check if there is any extension.
+    // Use the selected filter as the extension only if the filter has one type.
+    if (finfo.suffix().isEmpty() && selected_filter.count('*') == 1)
+    {
+        int start = selected_filter.indexOf("*") + 1;
+        int end = selected_filter.indexOf(")", start);
+
+        if (start != 0 && end > start)
```

Comment:
I think `start` cannot be 0 since it is guaranteed that there is one `*`.

Reply:
Let me use an insert then to be sure that the filter is correct